pyproject.toml

```diff
@@ -153,6 +153,7 @@ dependencies = [
     "types-python-dateutil",
     "Werkzeug<3.0.0",
     "dbt-core",
+    "methodtools",
```

Comment:
@pankajastro why do we need this dependency now?

Reply:
New common SQL provider uses this package, which is somehow missing. I feel might be this PR: https://github.com/apache/airflow/pull/41327/files causing it

Reply:
It feels like this dependency should be added to the provider - could you log a follow-up ticket in Airflow and link it in our code..?

Reply:
https://github.com/apache/airflow/issues/47147

Reply:
I also created a PR: https://github.com/apache/airflow/pull/47148